tests/h/views/activity_test.py

```diff
@@ -553,6 +553,64 @@ def test_search_passes_the_group_annotation_count_to_the_template(self,
         result = controller.search()['stats']
         assert result['annotation_count'] == 5
 
+    @pytest.mark.parametrize('test_group, test_user, test_heading, test_subtitle, test_share_msg',
+                             [('group', 'member', 'Members', 'Invite new members', 'Sharing the link lets people join this group:'),
+                              ('open_group', 'user', 'Moderators', 'Share group', 'Sharing the link lets people view this group:'),
+                              ('restricted_group', 'member', 'Contributors', 'Share group', 'Sharing the link lets people view this group:')],
+                             indirect=['test_group', 'test_user'])
+    def test_search_sets_display_strings_for_group(self,
+                                                   controller,
+                                                   test_group,
+                                                   test_user,
+                                                   test_heading,
+                                                   test_subtitle,
+                                                   test_share_msg,
+                                                   pyramid_request,
+                                                   search):
+
+        info = controller.search()['group_users_display']
+
+        assert info['heading'] == test_heading
+        assert info['subtitle'] == test_subtitle
+        assert info['share_msg'] == test_share_msg
+
+        search.reset_mock()
+
+    @pytest.mark.parametrize('test_group,test_user',
+                             [('group', 'member'),
+                              ('restricted_group', 'member')],
```

Comment:
I'm not sure why, but `restricted_group`'s `members` seem to be set to `None`

Reply:
Hi @sheetaluk : In the logic for the view controller, `members` is not being populated for groups that are `joinableBy` None: https://github.com/hypothesis/h/blob/965de34890bdd3140e575ada8b713b6cea9a8e3f/h/views/activity.py#L127

I believe that setting `joinable_by` to `None` in restricted groups is something we need to revisit—I don't think it's correct.

cc @hmstepanek